subprojects/core/src/main/java/org/gradle/internal/classpath/CallInterceptionClosureInstrumentingClassVisitor.java

```diff
@@ -72,7 +71,9 @@ private enum MethodInstrumentationStrategy {
         /**
          * Whenever the closure's delegate is set, we want to make sure that the call interception hooks are added to the new delegate's metaclass.
          */
-        SET_DELEGATE("setDelegate", getMethodDescriptor(Type.VOID_TYPE, getType(Object.class)), true, (classData, mv) -> {
+        SET_DELEGATE("setDelegate", getMethodDescriptor(Type.VOID_TYPE, getType(Object.class)), true) {
+            @Override
```

Comment:
https://errorprone.info/bugpattern/ImmutableEnumChecker

Reply:
this is the recommended way of fixing it though.
https://errorprone.info/bugpattern/ImmutableEnumChecker#:~:text=TIP%3A%20Instead%20of%20creating%20an%20enum%20with%20functional%20interface%20fields%20(Predicate%2C%20Function%2C%20etc.)%2C%20declare%20abstract%20methods%20that%20are%20overridden%20by%20each%20constant.%20For%20example%2C%20do%20this%3A